### PR TITLE
Assume room should be unencrypted if homeserver does not implement keys/query

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes https://github.com/vector-im/riot-web/issues/13598

### CODE REVIEW

File: src/createRoom.js [hunk 1]
```diff
@@ -212,7 +212,18 @@ export async function _waitForMember(client, roomId, userId, opts = { timeout: 1
...
+    try {
+        usersDeviceMap = await client.downloadKeys(userIds);
+    } catch (ex) {
+       if (ex.httpStatus === 404) {
+            console.log("Got 404 when fetching keys for users, not encrypting room");
+            // The endpoint to fetch keys doesn't exist: force unencrypted.
+            // See: https://github.com/vector-im/riot-web/issues/13598
+            return false;
+       }
+       throw ex;
```
Comment (turt2live) [inline src/createRoom.js:225]: We might just want to assume the user has no possible devices and instead return `false` (homeserver lacks support).

Comment (Half-Shot) [reply src/createRoom.js:225]: Yup, that was the original cut but wondering how much we care about things like matrix.org being flaky, triggering this exception, causing false positives.

Comment (turt2live) [reply src/createRoom.js:225]: Great question for product :)

**Comment by jryans** [comment]
@Half-Shot, are you planning to pursue the path Nad as outlined above?

**Comment by turt2live** [comment]
I'm not totally sure this is a thing that we should support (it's reasonable to expect that a server implements the endpoint), but curious to hear what other people think.

@Half-Shot is this still on your list?

**Comment by Half-Shot** [comment]
> I'm not totally sure this is a thing that we should support (it's reasonable to expect that a server implements the endpoint), but curious to hear what other people think.
> 
> @Half-Shot is this still on your list?

This was a long time ago back when Dendrite didn't do E2E and caused problems for us. It might not be worth perusing now.

**Comment by turt2live** [comment]
I'm going to close this for the time being then - if it's still an issue then folks will be able to find the code from the open issue.

Thanks!

**Pull request closed without merging**